map/src/main/kotlin/edu/artic/map/MapFragment.kt

```diff
@@ -392,6 +396,46 @@ class MapFragment : BaseViewModelFragment<MapViewModel>() {
                     refreshMapDisplayMode()
                 }
                 .disposedBy(disposeBag)
+
+        Observables.combineLatest(
+                viewModel.isUserInMuseum,
+                viewModel.currentMap
+        ).filter { (_, currentMap) -> currentMap.value != null }
+                .subscribe { (inMuseum, currentMap) ->
+                    val map = currentMap.value!!
+                    if (map.isMyLocationEnabled != inMuseum) {
+                        currentMap.value!!.isMyLocationEnabled = inMuseum
+                    }
+                }.disposedBy(disposeBag)
+
+        viewModel.showCompass
+                .bindToMain(compass.visibility())
+                .disposedBy(disposeBag)
+
+        viewModel.focusToTracking
+                .distinctUntilChanged()
+                .subscribe {(map, wrapped) ->
+                    if(wrapped.value != null) {
+                        compass.rotation = 0.0f
+                        compass.alpha = 1.0f
+                        map.uiSettings.setAllGesturesEnabled(false)
+                        val location = wrapped.value!!
+                        map.animateCamera(
+                                CameraUpdateFactory
+                                        .newCameraPosition(
+                                                CameraPosition.Builder(map.cameraPosition)
+                                                        .bearing(location.bearing)
+                                                        .target(LatLng(location.latitude, location.longitude))
+                                                        .build()
+                                        )
+                        )
+                    } else {
+                        map.uiSettings.setAllGesturesEnabled(true)
+                        compass.rotation = 30f
```

Comment:
To confirm, these should always be the same numbers as the ones in the compass `ImageView`'s LayoutParams?

Reply:
yes.